src/Controls/src/Core/Registrar.cs

```diff
@@ -79,6 +82,37 @@ public void Register(Type tview, Type trender, Type[] supportedVisuals, short pr
 		/// <include file="../../docs/Microsoft.Maui.Controls.Internals/Registrar.xml" path="//Member[@MemberName='Register']/Docs" />
 		public void Register(Type tview, Type trender) => Register(tview, trender, _defaultVisualRenderers);
 
+		HashSet<Type> _registeredVisuals = new HashSet<Type>();
+
+		internal void RegisterVisual([DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicParameterlessConstructor)] Type registeredVisual) => _registeredVisuals.Add(registeredVisual);
+
+		[return: DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicParameterlessConstructor)]
+		internal Type GetVisual(string visualName)
+		{
+			foreach (var visualType in _registeredVisuals)
+			{
+				string name = visualType.Name;
+				string fullName = visualType.FullName;
+
+				if (name.EndsWith("Visual", StringComparison.OrdinalIgnoreCase))
```

Comment:
It's odd that we use IgnoreCase here, but don't do IgnoreCase below when comparing `visualName == $"{name}Visual"`